### PR TITLE
Scatter forward projector setter

### LINKED ISSUE #1527: EstimateScatter forces the use of ForwardProjectorByBinUsingProjMatrixByBin for mask computation
Attempting to estimate scatter from a CT attenuation map, many small voxels. 

This leads to an issue with the `ForwardProjectorByBinUsingProjMatrixByBin` as https://github.com/UCL/STIR/blob/454be92288d3eac39bbaf79fd8bfdd1c71937521/src/recon_buildblock/DataSymmetriesForBins_PET_CartesianGrid.cxx#L74-L79

I am generally using parallelproj that avoids this issue elsewhere in the code base (no symmetries) but `estimate_scatter` forces the use of `ForwardProjectorByBinUsingProjMatrixByBin` when computing the mask projection data. 

https://github.com/UCL/STIR/blob/454be92288d3eac39bbaf79fd8bfdd1c71937521/src/scatter_buildblock/ScatterEstimation.cxx#L1238-L1240

_____

Two options
1. Pre-compute and set the mask projdata before setup is called.
2. Add a settable member forward projector shared pointer to the EstimateScatter class.

@KrisThielemans thoughts? Estimate scatter is already complex and convoluted, the second option adds more but simplifies. It can default to the current behavior.

### PR DESCRIPTION
**Pull request by robbietuk**
## Changes in this pull request

Introduces the ability to set and retrieve a forward projector for scatter estimation, used in mask forward projection, in the ScatterEstimation class. Also adds a keymap parser.

## Related issues

Fixes #1527

## Checklist before requesting a review

  - [x] I have performed a self-review of my code
  - [x] I have added docstrings/doxygen in line with the guidance in the developer guide
  - [] I have implemented unit tests that cover any new or modified functionality (if applicable)
  - [x] The code builds and runs on my machine
  - [x] `documentation/release_XXX.md` has been updated with any functionality change (if applicable)

### CODE REVIEW

**Comment by robbietuk** [comment]
Ready to merge.

File: src/include/stir/scatter/ScatterEstimation.h [hunk 1]
```diff
@@ -39,6 +39,7 @@
 
 #include "stir/stir_math.h"
 #include "stir/FilePath.h"
+#include <stir/recon_buildblock/ForwardProjectorByBin.h>
```
Comment (KrisThielemans) [inline src/include/stir/scatter/ScatterEstimation.h:42]:
```suggestion
#include "stir/recon_buildblock/ForwardProjectorByBin.h"
```

File: src/include/stir/scatter/ScatterEstimation.h [hunk 2]
```diff
@@ -170,6 +171,11 @@ class ScatterEstimation : public ParsingObject
   void set_recompute_mask_image(bool arg);
   void set_recompute_mask_projdata(bool arg);
 
+  //! Sets the forward projector for the scatter estimation
+  void set_forward_projector_sptr(const shared_ptr<ForwardProjectorByBin> projector_sptr);
+  //!Get the forward projector used for the scatter estimation
+  shared_ptr<ForwardProjectorByBin> get_forward_projector_sptr() const;
```
Comment (KrisThielemans) [inline src/include/stir/scatter/ScatterEstimation.h:177]: need to change naming. These are currently only used for the sino mask generation. Maybe `....forward_projector_for_mask...`?

File: src/include/stir/scatter/ScatterEstimation.h [hunk 3]
```diff
@@ -275,6 +281,8 @@ class ScatterEstimation : public ParsingObject
   //! The file name for the attenuation coefficients.
   //! If they are to be recalculated they will be stored here, if set.
   std::string atten_coeff_filename;
+  //! ForwardProjector
+  shared_ptr<ForwardProjectorByBin> forward_projector_sptr;
```
Comment (KrisThielemans) [inline src/include/stir/scatter/ScatterEstimation.h:285]: name here as well

File: src/scatter_buildblock/ScatterEstimation.cxx [hunk 4]
```diff
@@ -112,6 +112,10 @@ ScatterEstimation::initialise_keymap()
   this->parser.add_key("mask projdata filename", &this->mask_projdata_filename);
   this->parser.add_key("tail fitting parameter filename", &this->tail_mask_par_filename);
   // END MASK
+
+  // Forward projector for mask projection
+  this->parser.add_parsing_key("forward projector type", &this->forward_projector_sptr);
```
Comment (KrisThielemans) [inline src/scatter_buildblock/ScatterEstimation.cxx:117]: and here

**Review by KrisThielemans — CHANGES_REQUESTED** [review]

**Review by KrisThielemans — APPROVED** [review]

**Comment by KrisThielemans** [comment]
ah, sorry, can you add a line to `release_6.3.htm` please? (include `[ci skip]` in the commit, as it seems our "exclude `*htm` doesn't work for GitHub Actions)

**Comment by KrisThielemans** [comment]
I'll squash-merge if ok

**Comment by KrisThielemans** [comment]
looks like there's a conflict with `master` in the release notes...

**Pull request merged**